FPEngine.cc

```diff
@@ -479,6 +490,25 @@ void FPNetworkControl::probe_transmission_handler(nsock_pool nsp, nsock_event ns
         myprobe->host->fail_one_probe();
         gh_perror("Unable to send packet in %s", __func__);
       }
+      /* If Packet was sent successfully and Decoys are present */
+      else if (o.numdecoys > 1) {  
+        FPPacket *temp = new FPPacket();
+        std::map<std::string, IPv6Header*>::iterator it;
+        char addr[INET6_ADDRSTRLEN];
+        for (int decoy = 0; decoy < o.numdecoys; decoy++) {
+          if (decoy == o.decoyturn)
+            continue;
+          sockaddr_storage_iptop(myprobe->host->getTargetAddress(), addr);
```

Comment:
Even if we stick with the map structure, no need to use printable IP address as the key if it is never printed. We know it will always be an IPv6 address, so why not use the `sin6_addr` element directly?